### PR TITLE
Improve the build.sh

### PR DESCRIPTION
**Pull request by lawrencelomax**
Adds archiving and uses `xcodebuild` for building, so this should help out the homebrew/deploy case.

### CODE REVIEW

File: build.sh [hunk 1]
```diff
@@ -1,76 +1,142 @@
...
+}
+
+if [[ -n $TARGET ]]; then
+  echo "using target $TARGET"
+elif [[ -n $1 ]]; then
+  echo "using target $1"
+  TARGET=$1
 else
-  echo "Invalid mode $MODE"
+  echo 'No target argument or $TARGET provided'
```
Comment (asm89) [inline build.sh]: You need " for the expression to expand.

File: build.sh [hunk 2]
```diff
@@ -1,76 +1,142 @@
...
   exit 1
 fi
 
+if [[ -n $COMMAND ]]; then
+  echo "using command $COMMAND"
+elif [[ -n $2 ]]; then
+  echo "using target $2"
+  COMMAND=$2
+else 
+  echo 'No command argument or $COMMAND provided'
```
Comment (asm89) [inline build.sh]: Same here.

File: build.sh [hunk 3]
```diff
@@ -1,76 +1,142 @@
...
-  echo "Invalid mode $MODE"
+  echo 'No target argument or $TARGET provided'
   print_usage
   exit 1
 fi
 
+if [[ -n $COMMAND ]]; then
+  echo "using command $COMMAND"
+elif [[ -n $2 ]]; then
+  echo "using target $2"
```
Comment (phyllipecesar) [inline build.sh]: using command!

**Comment by asm89** [comment]
nits + this starts to become python material imo

**Comment by lawrencelomax** [comment]
@facebook-github-bot shipit

**Pull request merged**